components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/BasicAuthenticationHandler.java

```diff
@@ -108,11 +108,11 @@ protected AuthenticationResult doAuthenticate(MessageContext messageContext) thr
                     //TODO: Related to this https://wso2.org/jira/browse/IDENTITY-4752 - Class IdentityMgtEventListener
                     // : Line 563: Have to check whether why we can't continue
                     //without following lines as previous code.
-                    PrivilegedCarbonContext.startTenantFlow();
-                    PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain);
-                    PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
-
                     try {
+                        PrivilegedCarbonContext.startTenantFlow();
```

Comment:
Start tenant flow should be immediately before the try block to make sure that end tenant flow is executed in the finally block only when the start tenant flow is successful.